de.bund.bfr.knime.fsklab.r/src/de/bund/bfr/knime/fsklab/r/client/LibRegistry.java

```diff
@@ -135,7 +126,7 @@ private LibRegistry() throws IOException, RException {
   }
 
   public synchronized static LibRegistry instance() throws IOException, RException {
-    if (instance == null || PreferenceInitializer.refresh) {
```

Comment:
@schuelet: removing this check will prevent the recreation of the LibRegistry instance, which is needed when the user changes the global FSK setting(this is part of RProfile managment)
why it's necessary to remove the condition?

Reply:
I see you description now, but have you tested it with changing the global setting?

Reply:
> @schuelet: removing this check will prevent the recreation of the LibRegistry instance, which is needed when the user changes the global FSK setting(this is part of RProfile managment)
> why it's necessary to remove the condition?

@ahmadswaid Each time an RController is created, PreferenceInitializer.refresh is set to true. See RController.initR() ->
`m_rProps =RBinUtil.retrieveRProperties();` . Therefore, each time a new LibRegistry instance will be created.

I do see your point, so I will try to find a workaround.